test/service/test_health_check.py

```diff
@@ -48,17 +48,11 @@ def test_all_api_endpoints_down(self):
                                                 down_endpoints=self.endpoints)
         response = self._test(endpoint_states, lambdas_up=True)
         self.assertEqual(503, response.status_code)
-        self.assertEqual(self._expected_health(endpoint_states), response.json())
-
-    @mock_sts
-    @mock_sqs
-    def test_one_api_endpoint_down(self):
-        self._create_mock_queues()
-        endpoint_states = self._endpoint_states(up_endpoints=self.endpoints[1:],
-                                                down_endpoints=self.endpoints[:1])
-        response = self._test(endpoint_states, lambdas_up=True)
-        self.assertEqual(503, response.status_code)
-        self.assertEqual(self._expected_health(endpoint_states), response.json())
+        actual_health = response.json()
+        endpoint_states = {
+            e: u for e, u in endpoint_states.items() if e in actual_health['api_endpoints']['error']
+        }
+        self.assertEqual(self._expected_health(endpoint_states), actual_health)
```

Comment:
This is convoluted. You're deriving the expectation from the actual outcome, which is a smell. You've simplified the response, so the expectation should be simpler, too. It seems like the only variability is in the URL in the `error` property of the response. I see two ways to deal with that: fingerprinting or a pattern match.